File: src/Lesson6/Java/Ex4.java
package Lesson6.Java;

public class Ex4 {
    public static void main(String[] args) {

        // Исходный массив
        int[] arr = {4, 2, 2, 8, 5, 6, 8, 5};

        int[] unique = new int[arr.length];

        int uniqueCount = 0;

        for (int i = 0; i < arr.length; i++) {

            boolean alreadyExists = false;

            for (int j = 0; j < uniqueCount; j++) {

                if (arr[i] == unique[j]) {

                    alreadyExists = true;
                    break;
                }
            }

            if (!alreadyExists) {

                unique[uniqueCount] = arr[i];

                uniqueCount++;
            }
        }

        for (int i = 0; i < uniqueCount; i++) {

            for (int j = 0; j < uniqueCount - 1 - i; j++) {

                if (unique[j] > unique[j + 1]) {

                    int temp = unique[j];
                    unique[j] = unique[j + 1];
                    unique[j + 1] = temp;
                }
            }
        }

        for (int i = 0; i < uniqueCount; i++) {

            System.out.print(unique[i] + " ");
        }
    }
}

File: src/Lesson6/forkotlin/Ex4.kt
package Lesson6.forkotlin

fun main() {
    val arr = intArrayOf(4, 2, 2, 8, 5, 6, 8, 5)

    val unique = IntArray(arr.size)
    var uniqueCount = 0

    for (i in arr.indices) {
        var alreadyExists = false

        for (j in 0 until uniqueCount) {
            if (arr[i] == unique[j]) {
                alreadyExists = true
                break
            }
        }

        if (!alreadyExists) {
            unique[uniqueCount] = arr[i]
            uniqueCount++
        }
    }

    for (i in 0 until uniqueCount) {
        for (j in 0 until uniqueCount - 1 - i) {
            if (unique[j] > unique[j + 1]) {
                val temp = unique[j]
                unique[j] = unique[j + 1]
                unique[j + 1] = temp
            }
        }
    }

    for (i in 0 until uniqueCount) {
        print("${unique[i]} ")
    }
}